outbound.js

```diff
@@ -138,69 +141,68 @@
 
         var typeofUserId = typeof userId;
         if (typeofUserId != "number" && typeofUserId != "string") {
             deferred.reject(error("Invalid user ID. Expected string or number, got " + typeofUserId, false));
         } else {
-            requestData = {"user_id": userId}
-
-            user = userObject(info, attributes)
+            requestData = {"user_id": userId};
+            user = userObject(attributes);
             for (var attr in user) {
                 requestData[attr] = user[attr];
             }
-
             post('/identify', requestData, deferred);
         }
         return deferred.promise;
-    }
+    };
 
-    Outbound.prototype.track = function(userId, event, properties, userInfo, userAttributes) {
+    Outbound.prototype.track = function(userId, event, properties, timestamp) {
         var deferred = D();
 
         var typeofUserId = typeof userId;
         var typeofEvent = typeof event;
+
         if (typeofUserId != "number" && typeofUserId != "string") {
             deferred.reject(error("Invalid user ID. Expected string or number, got " + typeofUserId, false));
-        } else  if (typeofEvent != "string") {
+        } else if (typeofEvent != "string") {
             deferred.reject(error("Invalid event. Expected string, got " + typeofEvent, false));
         } else {
-            requestData = {"user_id": userId, "properties": {}, "event": event}
+            requestData = {"user_id": userId, "properties": {}, "event": event};
 
-            user = userObject(userInfo, userAttributes)
-            for (var attr in user) {
-                if (!requestData.user) {
-                    requestData.user = {};
-                }
-                requestData.user[attr] = user[attr];
+            if (timestamp){
+              requestData.timestamp = timestamp;
+            } else {
+              requestData.timestamp = Math.floor(Date.now() / 1000);
             }
 
             if (properties && typeof properties === 'object') {
                 requestData.properties = properties;
             }
+
             post('/track', requestData, deferred);
         }
+
         return deferred.promise;
-    }
+    };
 
     Outbound.prototype.registerApnsToken = function(userId, token) {
         return deviceToken(APNS, true, userId, token);
-    }
+    };
 
     Outbound.prototype.registerGcmToken = function(userId, token) {
         return deviceToken(GCM, true, userId, token);
-    }
+    };
 
     Outbound.prototype.disableApnsToken = function(userId, token) {
         return deviceToken(APNS, false, userId, token);
-    }
+    };
 
     Outbound.prototype.disableGcmToken = function(userId, token) {
         return deviceToken(GCM, false, userId, token);
-    }
+    };
 
-    typeof window !== 'undefined' && (window.outbound = Outbound)
-    typeof module !== 'undefined' && module.exports && (module.exports = Outbound)
+    typeof window !== 'undefined' && (window.outbound = Outbound);
```

Comment:
You can remove this line and change the next line to just `module.exports = Outbound`. I put this in here for both browser and node module support which we don't need anymore.